GPTutils.py

```diff
@@ -0,0 +1,262 @@
+from typing import List, Dict, Tuple, Union
+import asyncio
+
+from openai import OpenAI
+import openai
+import langchain
+from langchain.embeddings import OpenAIEmbeddings
+from langchain.callbacks.streaming_aiter_final_only import (
+    AsyncFinalIteratorCallbackHandler,
+)
+from langchain.text_splitter import (
+    RecursiveCharacterTextSplitter,
+    CharacterTextSplitter,
+)
+import numpy as np
+from numpy.linalg import norm
+from GPTsettings import GPTsettings
+
+mapper = {
+    "py": "python",
+    "js": "js",
+    "jsx": "js",
+    "ts": "ts",
+    "java": "java",
+    "html": "html",
+    "md": "markdown",
+}
+
+
+class GPTWrapper:
+    def __init__(self, model: str = GPTsettings.MODEL) -> None:
+        """
+        Initializes a new instance of the GPTWrapper class.
```

Comment:
Missing docstrings for new functions and classes, please add them to maintain proper documentation.